src/basic/namespace-util.h

```diff
@@ -3,6 +3,24 @@
 
 #include <sys/types.h>
 
+typedef enum namespace_type {
+        USER_NS   = 0,
+        MNT_NS    = 1,
+        PID_NS    = 2,
+        UTS_NS    = 3,
+        IPC_NS    = 4,
+        NET_NS    = 5,
+        CGROUP_NS = 6,
+        TIME_NS   = 7,
+        MAX_NS    = 8
```

Comment:
throughout our codebase, we usually have namespace prefixes, not suffixes. or does this mirror some in-kernel definitions that have these names?

any reason the values are explicitly specified?

we usually put a trailing `,` after the last enum value, to minimize size of later patches...

For most our enums we define things like this btw:

```c
typedef enum NamespaceType {
        NAMESPACE_USER,
        NAMESPACE_MNT,
        NAMESPACE_PID, 
        …
        NAMESPACE_TIME,
        _NAMESPACE_TYPE_MAX,
        _NAMESPACE_TYPE_INVALID = -EINVAL,
} NamespaceType;
``

i.e. we add an `_` prefix to the entries that aren't real values for the enum.

Reply:
Specifying enum values explicitly makes it hard to mess them up when e.g., inserting new enums in the middle and other unintentional bugs.